components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -339,12 +341,14 @@ pub struct ReadStats {
     // RegionID -> RegionInfo
     pub region_infos: HashMap<u64, RegionInfo>,
     pub sample_num: usize,
+    pub region_buckets: HashMap<u64, BucketStat>,
```

Comment:
So now we store BucketStat in two places, one is in fsm.peer for write and region_buckets here for read. 
And actually ReadStats is TLS which means we will have one region_buckets per thread. 
My question is when buckets are changed (e.g. split/merge), how can all these ReadStats instances get updated with the new buckets.  Today only fsm.peer's region_buckets get updated.  Is it possible to read fsm.peer's region_bucket from here?

Reply:
Reading fsm.peer's region_bucket needs coprocessor to sync with raftstore and I'm concerned it could affect the performance. Buckets in ReadStats can be updated when collecting read flow.